src/keyboard/mod.rs

```diff
@@ -625,3 +657,311 @@ where
         },
     )
 }
+
+/// Implement a keyboard to automatically detect the keymap and send KeyRepeatEvents
+/// at set intervals
+///
+/// This requires you to provide an implementation to receive the events after they
+/// have been interpreted with the keymap, as well as an implementation to be called
+/// when KeyRepeatEvents are sent at intervals set by the KeyRepeatKind argument.
+///
+/// The keymap information will be loaded from the events sent by the compositor,
+/// as such you need to call this method as soon as you have created the keyboard
+/// to make sure this event does not get lost.
+///
+/// Returns an error if xkbcommon could not be initialized.
+pub fn map_keyboard_auto_with_repeat<Impl, RepeatImpl>(
+    keyboard: NewProxy<wl_keyboard::WlKeyboard>,
+    key_repeat_kind: KeyRepeatKind,
+    implementation: Impl,
+    repeat_implementation: RepeatImpl,
+) -> Result<Proxy<wl_keyboard::WlKeyboard>, (Error, NewProxy<wl_keyboard::WlKeyboard>)>
+where
+    for<'a> Impl: Implementation<Proxy<wl_keyboard::WlKeyboard>, Event<'a>> + Send,
+    RepeatImpl: Fn(KeyRepeatEvent) + Send + 'static,
```

Comment:
Please use the `Implementation` trait here as well, so

`RepeatImpl: Implementation<Proxy<wl_keyboard::WlKeyboard>, KeyRepeatEvent> + Send`